addons/account_peppol/wizard/peppol_registration.py

```diff
@@ -0,0 +1,272 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models, modules
+from odoo.exceptions import UserError, ValidationError
+
+from odoo.addons.account_peppol.tools.demo_utils import handle_demo
+
+
+class PeppolRegistration(models.TransientModel):
+    _name = 'peppol.registration'
+    _description = "Peppol Registration"
+
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        required=True,
+        default=lambda self: self.env.company,
+    )
+    account_peppol_contact_email = fields.Char(
+        related='company_id.account_peppol_contact_email', readonly=False,
+        required=True,
+    )
+    account_peppol_edi_mode = fields.Selection(
+        string='EDI mode',
+        selection=[('demo', 'Demo'), ('test', 'Test'), ('prod', 'Live')],
+        compute='_compute_account_peppol_edi_mode',
+        inverse='_inverse_account_peppol_edi_mode',
+        readonly=False,
+    )
+    account_peppol_edi_user_id = fields.Many2one(
+        comodel_name='account_edi_proxy_client.user',
+        string='EDI user',
+        compute='_compute_account_peppol_edi_user_id',
+    )
+    account_peppol_migration_key = fields.Char(related='company_id.account_peppol_migration_key', readonly=False)
+    account_peppol_mode_constraint = fields.Selection(
+        selection=[('demo', 'Demo'), ('test', 'Test'), ('prod', 'Live')],
+        compute='_compute_account_peppol_mode_constraint',
+        help="Using the config params, this field specifies which edi modes may be selected from the UI"
+    )
+    account_peppol_phone_number = fields.Char(
```

Comment:
there is a bug in stable with the related field:
```py
    @api.depends('phone')
    def _compute_account_peppol_phone_number(self):
        for company in self:
            if not company.account_peppol_phone_number:
                try:
                    # precompute only if it's a valid phone number
                    company._sanitize_peppol_phone_number(company.phone)
                    company.account_peppol_phone_number = company.phone
                except ValidationError:
                    continue
```
... the peppol phone number is never recomputed after it get a value even when we change the phone number of the company after that

Reply:
Also, the phone number should be always prefixed by the country prefix (e.g. +32 for BE). Can you add it automatically plz?

Reply:
When setting the wrong phone number, the user error is opened twice on the UI... it's weird

Reply:
do we really want to recompute phone number they provided for peppol? not sure. we need to keep whichever phone number they verified I think?